### PR TITLE
Bala/Add comment on failure

### PR DESCRIPTION
**Pull request by balakrishna-deriv**
Changes:

-   Add GitHub action comment on build.
-   Add GitHub action comment on failure.

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: .github/workflows/generate-preview-link.yml [hunk 1]
```diff
@@ -112,22 +139,38 @@ jobs:
...
-                          `**Preview Link**: ${preview_url}`,
-                          '| Name | Result |',
-                          '| :--- | :------ |',
-                          `| **Build status**  | Completed ✅ |`,
-                          `| **Preview URL**  | [Visit Preview](${preview_url}) |`,
-                          ''
-                        ].join('\n')
+                      let comment;
+                      const action_url = "${{github.server_url}}/${{github.repository}}/actions/runs/${{github.run_id}}"
+                      const status = "${{steps.publish-to-pages.outcome}}"
```
Comment (adrienne-deriv) [inline .github/workflows/generate-preview-link.yml:150]: If the build step fails should we also show this comment?

Comment (adrienne-deriv) [reply .github/workflows/generate-preview-link.yml:150]: Imo maybe we can separate the steps, like:
```suggestion
      - name: 'Generate failure comment'
        if: failure()
        ....generate fail comment
        
      - name: 'Generate preview link comment'
         if: success()
         ...generate success comment
        
```

Comment (adrienne-deriv) [reply .github/workflows/generate-preview-link.yml:150]: Screenshot 2023-09-27 at 1 00 48 PM

Comment (balakrishna-deriv) [reply .github/workflows/generate-preview-link.yml]: Done. Thanks for the review @adrienne-deriv 🙏

**Review by adrienne-deriv — APPROVED** [review]
LGTM

**Review by habib-deriv — APPROVED** [review]

**Pull request merged**